src/main/kotlin/no/nav/familie/klage/integrasjoner/FagsystemVedtakService.kt

```diff
@@ -52,23 +57,34 @@ class FagsystemVedtakService(
         val fagsak = fagsakService.hentFagsakForBehandling(behandlingId)
         return when (fagsak.fagsystem) {
             Fagsystem.EF -> familieEFSakClient.kanOppretteRevurdering(fagsak.eksternId)
+            Fagsystem.KS -> familieKSSakClient.kanOppretteRevurdering(fagsak.eksternId)
             else -> throw Feil("Ikke implementert sjekk for å opprette revurdering for fagsak=${fagsak.id} fagsystem=${fagsak.fagsystem}")
         }
     }
 
     fun opprettRevurdering(behandlingId: UUID): OpprettRevurderingResponse {
         val fagsak = fagsakService.hentFagsakForBehandling(behandlingId)
-        return when (fagsak.fagsystem) {
-            Fagsystem.EF -> try {
-                familieEFSakClient.opprettRevurdering(fagsak.eksternId)
-            } catch (e: Exception) {
-                val errorSuffix = "Feilet opprettelse av revurdering for behandling=$behandlingId eksternFagsakId=${fagsak.eksternId}"
-                logger.warn("$errorSuffix, se detaljer i secureLogs")
-                secureLogger.warn(errorSuffix, e)
+        val behandling = behandlingRepository.findByIdOrThrow(behandlingId)
+        return try {
+            when (fagsak.fagsystem) {
+                Fagsystem.EF -> familieEFSakClient.opprettRevurdering(fagsak.eksternId)
+
+                Fagsystem.KS -> {
+                    val eksternFagsystemBehandlingId =
+                        behandling.påklagetVedtak.påklagetVedtakDetaljer?.eksternFagsystemBehandlingId
+                            ?: throw Feil("eksternFagsystemBehandlingId er ikke satt på påklagetVedtak for ks-behandling")
```

Comment:
Trenger dere ekstern behandlingId for å opprette revurdering automatisk etter omgjøring?
Dersom noen klager på en tilbakekreving så har du ikke nødvendigvis en fagsystembehandlingId tilgjengelig?

Reply:
I hvilke tilfeller er det vi ikke har det? Vi trenger det for å vite om behandlingen gjelder EØS eller nasjonale regler

Reply:
For vår del;
- Tilbakekrevingssaker som gjelder infotrygd har null her
- Utestengelser har null her (work in progress)
- Hvis det velges en tilbakekreving så peker denne verdien på tilbakekreving sin behandling, ikke fagsystemet sin

Men hvis dere krever at det finnes, og ikke minst MÅ gjelde fagsystemets behandling, så bør det også komme med i sjekken om man "kan" opprette revurdering automatisk.

Reply:
Enig! Fikser i [Bruk fagsak for å opprette revurdeing i KS-sak](https://github.com/navikt/familie-klage/pull/224/commits/c508e1f11bb97b926ff118424c9b8465bbf78b40)